xinput.odin

```diff
@@ -1,285 +1,263 @@
+XInputGetBatteryInformationStub : _XInputGetBatteryInformation : proc "stdcall" (dwUserIndex : u32,
+                                                                                 devType : Battery_Devtype,
+                                                                                 pBatteryInformation : ^XINPUT_BATTERY_INFORMATION)  -> (Error : u32) {
+    return 1167; //ERROR_DEVICE_NOT_CONNECTED
+}
+_XInputGetCapabilities       :: distinct #type proc "stdcall" (u32, u32, ^XINPUT_CAPABILITIES) -> u32;
+XInputGetCapabilitiesStub : _XInputGetCapabilities : proc "stdcall" (dwUserIndex : u32,
+                                                                     dwFlags : u32,
+                                                                     pCapabilities : ^XINPUT_CAPABILITIES)  -> (Error : u32) {
+    return 1167; //ERROR_DEVICE_NOT_CONNECTED
+}
+// TODO: XInputGetDSoundAudioDeviceGuids - https://docs.microsoft.com/en-us/windows/win32/api/xinput/nf-xinput-xinputgetdsoundaudiodeviceguids
+_XInputGetKeystroke          :: distinct #type proc "stdcall" (u32, u32, ^XINPUT_KEYSTROKE) -> u32;
+XInputGetKeystrokeStub : _XInputGetKeystroke : proc "stdcall" (dwUserIndex : u32,
+                                                               dwReserved : u32,
+                                                               pKeystroke : ^XINPUT_KEYSTROKE)  -> (Error : u32) {
+    return 1167; //ERROR_DEVICE_NOT_CONNECTED
+}
+_XInputGetState              :: distinct #type proc "stdcall" (u32, ^XINPUT_STATE) -> u32;
+XInputGetStateStub : _XInputGetState : proc "stdcall" (dwUserIndex : u32,
+                                                       pState : ^XINPUT_STATE)  -> (Error : u32) {
+    return 1167; //ERROR_DEVICE_NOT_CONNECTED
+}
+_XInputSetState              :: distinct #type proc "stdcall" (u32, ^XINPUT_VIBRATION) -> u32;
+XInputSetStateStub : _XInputSetState : proc "stdcall" (dwUserIndex : u32,
+                                                       pVibration : ^XINPUT_VIBRATION)  -> (Error : u32) {
+    return 1167; //ERROR_DEVICE_NOT_CONNECTED
+}
 
-Version := XInputVersion.NotLoaded;
-
-set_proc_address :: #type proc(lib : rawptr, p: rawptr, name: string) #inline;
-load_library     :: #type proc(name : string) -> rawptr;
-
-init :: proc(set_proc : set_proc_address, load_lib : load_library, initalState : bool = true) -> bool {
-    lib := load_lib("xinput1_4.dll"); //NOTE(Hoej): Freeing this will make any xinput calls panic at runtime.
-    using XInputVersion;
+XInputEnable                := XInputEnableStub;
+XInputGetBatteryInformation := XInputGetBatteryInformationStub;
+XInputGetCapabilities       := XInputGetCapabilitiesStub;
+XInputGetKeystroke          := XInputGetKeystrokeStub;
+XInputGetState              := XInputGetStateStub;
+XInputSetState              := XInputSetStateStub;
+ 
+init :: proc(initialState : bool = true) -> bool {
+    xinput_library := win32.load_library_a("XInput1_4.dll");
+    using XInput_Version;
```

Comment:
Please use the full qualifier the enums instead of using on enums (using outside structs fields I am generally not a fan of)